xls/modules/zstd/memory/BUILD

```diff
@@ -0,0 +1,675 @@
+# Copyright 2024 The XLS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@xls_pip_deps//:requirements.bzl", "requirement")
+load("@rules_hdl//place_and_route:build_defs.bzl", "place_and_route")
+load("@rules_hdl//synthesis:build_defs.bzl", "benchmark_synth", "synthesize_rtl")
+load("@rules_hdl//verilog:providers.bzl", "verilog_library")
+load(
+    "//xls/build_rules:xls_build_defs.bzl",
+    "xls_benchmark_ir",
+    "xls_benchmark_verilog",
+    "xls_dslx_library",
+    "xls_dslx_test",
+    "xls_dslx_verilog",
+)
+
+package(
+    default_applicable_licenses = ["//:license"],
+    default_visibility = ["//xls:xls_users"],
+    licenses = ["notice"],
+)
+
+xls_dslx_library(
+    name = "axi_dslx",
+    srcs = ["axi.x"],
+)
+
+xls_dslx_library(
+    name = "axi_st_dslx",
+    srcs = ["axi_st.x"],
+)
+
+xls_dslx_library(
+    name = "common_dslx",
+    srcs = [
+        "common.x",
+    ],
+    deps = [":axi_dslx"],
+)
+
+xls_dslx_test(
+    name = "common_dslx_test",
+    library = ":common_dslx",
+    tags = ["manual"],
+)
+
+CLOCK_PERIOD_PS = "750"
+
+common_codegen_args = {
+    "delay_model": "asap7",
+    "reset": "rst",
+    "worst_case_throughput": "1",
+    "use_system_verilog": "false",
+    "clock_period_ps": CLOCK_PERIOD_PS,
+    "clock_margin_percent": "20",
+    "multi_proc": "true",
+}
+
+xls_dslx_library(
+    name = "axi_reader_dslx",
+    srcs = ["axi_reader.x"],
+    deps = [
+        ":axi_dslx",
+        ":axi_st_dslx",
+        ":common_dslx",
+    ],
+)
+
+xls_dslx_test(
+    name = "axi_reader_dslx_test",
+    library = ":axi_reader_dslx",
+    tags = ["manual"],
+)
+
+axi_reader_codegen_args = common_codegen_args | {
+    "module_name": "axi_reader",
+    "pipeline_stages": "2",
+    "streaming_channel_data_suffix": "_data",
+    "flop_inputs_kind": "skid",
+    "flop_outputs_kind": "skid",
+    "clock_period_ps": "1800",
```

Comment:
any reason the clock period is different here? should we keep a list of those at the top for easier tuning?

Reply:
The critical path delay from the IR benchmark for this proc exceeds the 750ps that we use throughout the ZSTD Decoder. In order to provide valid constraints for the scheduler we increased the clock period to the value that is close to the critical path delay. Moved this to a variable at the top of the file where the `750ps` clock period is also defined.